File: app/src/main/java/com/lucas/omnia/fragments/RecentPostsTabFragment.java
package com.lucas.omnia.fragments;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;

public class RecentPostsTabFragment extends PostListFragment {

    public RecentPostsTabFragment() {
        // Required empty public constructor
    }

    @Override
    public Query getQuery(DatabaseReference databaseReference) {
        // Last 100 posts
        return databaseReference.child("posts").limitToFirst(100);
    }
}

File: app/src/main/java/com/lucas/omnia/fragments/RecentPostsTabFragment.kt
package com.lucas.omnia.fragments

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

class RecentPostsTabFragment : PostListFragment() {
    override fun getQuery(databaseReference: DatabaseReference?): Query {
        // Last 100 posts
        return databaseReference!!.child("posts").limitToFirst(100)
    }
}